ui/src/app/workflows/components/workflows-list/workflows-list.tsx

```diff
@@ -35,21 +38,14 @@ export class WorkflowsList extends BasePage<RouteComponentProps<any>, State> {
 
     private set namespace(namespace: string) {
         this.setState({namespace});
-        history.pushState(null, '', uiUrl('workflows/' + namespace));
-        this.fetchWorkflows();
```

Comment:
These shouldn't be blindly removed. If the namespace is changed with `this.namespace = <...>` without calling `handleChanges` (as is done here https://github.com/argoproj/argo/pull/2196/files#diff-65a748d49e1d0e37a029767481ed3487R138) then the workflows won't be re-fetched.

Currently we change the namespace with one of three ways: using `this.setState({namespace})`, calling `handleChanges`, or setting directly with `this.namespace = <...>`. We should only use one of them and consolidate the rest into it. Also we should make sure that all of the state properties are changed with this single method.

Reply:
Should not add any logic here.

In regards to https://github.com/argoproj/argo/pull/2196/files#diff-65a748d49e1d0e37a029767481ed3487R138, it's already in function `fetchWorkflows`, followed by a `service.list` call to fetch the workflow list. 

And think about the case when `this.namespace = info.managedNamespace;` is invoked, only when the server is running on managed namespace mode (let's say in ns `argo`), and the user opens the url `/workflows/` or `/workflows/abc`, that code will be hit. In this case, the `namespace` in the state is changed to `argo`, so workflows in the page content is correct. The only defect is - the url shows `/workflows/abc` but the page shows namespace `argo`. However, we can not simply add `history.pushState(null, '', uiUrl('workflows/' + namespace))` to fix it, because user might hit the url `/workflows/abc/phase=Runningxxxxx`, we need to plus the queries in the url . It's not perfect in any case, so I prefer not to add it.